javascript/ql/src/Security/CWE-598/SensitiveGetQuery.ql

```diff
@@ -0,0 +1,27 @@
+/**
+ * @name Sensitive data read from GET request
+ * @description Placing sensitive data in a GET request increases the risk of
+ *              the data being exposed to an attacker.
+ * @kind problem
+ * @problem.severity warning
+ * @security-severity 6.5
+ * @precision high
+ * @id js/sensitive-get-query
+ * @tags security
+ *       external/cwe/cwe-598
+ */
+
+import javascript
+
+from
+  Express::RouteSetup setup, Express::RouteHandler handler, Express::RequestInputAccess input,
+  SensitiveExpr sensitive
+where
+  setup.getRequestMethod() = "GET" and
+  handler = setup.getARouteHandler() and
+  input.getRouteHandler() = handler and
+  input.getKind() = "parameter" and
+  input.(DataFlow::SourceNode).flowsToExpr(sensitive) and
```

Comment:
Lets discuss the expressivity in this thread. What is the reason we are not using a full Taint/DataFlow configuration here?

Alternative with slightly more reach than just `flowsToExpr`: Could we use (smallstep? intraprocedural?) type tracking?

Reply:
I made that choice for simplicity. Both simplicity of the query, and simplicity of the alert message.  
And I don't think we loose that many results. 

I think the `RequestInputAccess` class gives us the tracking we need for this query.  
`RequestInputAccess` gives us a read of a specific query parameter (e.g. `reg.query.password`).  
The test actually works just fine with just `input.asExpr() = sensitive` (I should add another test).